xarray/core/dataset.py

```diff
@@ -519,7 +535,7 @@ def __deepcopy__(self, memo=None):
     @property
     def _attr_sources(self):
         """List of places to look-up items for attribute-style access"""
-        return [self, LevelCoordinatesSource(self), self.attrs]
+        return [self, LevelCoordinatesSource(self), self.attrs, self.encoding]
```

Comment:
I don't think this is a good idea. This lets you write things like `dataset.unlimited_dims` to pull out unlimited dimensions from `encoding`, but `dataset.encoding['unlimited_dims']` is more explicit and this runs the risk of confusing `encoding`/`attrs`.